src/gui/tray/UserModel.cpp

```diff
@@ -496,11 +499,26 @@ void User::slotItemCompleted(const QString &folder, const SyncFileItemPtr &item)
             _activityModel->addIgnoredFileToList(activity);
         } else {
             // add 'protocol error' to activity list
+            if (item->_status == SyncFileItem::Status::FileNameInvalid) {
+                displayNotification(item->_file, activity._subject);
+            }
             _activityModel->addErrorToActivityList(activity);
```

Comment:
@FlexW Do you still need to call **addErrorToActivityList** if the previous condition is **true**?

Reply:
@FlexW Or, is **displayNotification** only displays a notification to the OS native popup? Like, near tray on Windows? In that case, I'd rename this function to something more meaningful. Like, **showOSNotificationPopup**. Or, whatever, as long as it is clear from the function's name that it's triggering the OS to display a notification via native UI.

Reply:
@allexzander It's still necessary. displayNotification just displays os notifications. I would then go for `showOsNotification()`. Is that ok for you?